web/yo/app/scripts/directives/reviewpanel.js

```diff
@@ -211,7 +212,17 @@ angular.module('oncokbApp')
                             var historyData = [getEvidenceResult.historyData];
                             DatabaseConnector.updateEvidenceBatch(evidences, historyData, function(result) {
                                 $scope.modelUpdate($scope.adjustedEvidenceType, $scope.mutation, $scope.tumor, $scope.therapyCategory, $scope.treatment);
-                                ReviewResource.loading = _.without(ReviewResource.loading, $scope.uuid.getText());
+                                if ($scope.adjustedEvidenceType === 'TREATMENT_NAME_CHANGE' && _.isFunction($scope.updatePriorityInGene)) {
```

Comment:
add a corresponding ELSE block and put the statement below inside. Otherwise for non treatment name change, it will never finish loading
ReviewResource.loading = _.without(ReviewResource.loading, $scope.uuid.getText());